migrations/Version202203021737164734_taoBackOffice.php

```diff
@@ -0,0 +1,162 @@
+                'test/item/BRS' => [
+                    'taoQtiTest/controller/creator/views/item' => ['BRS'],
+                ],
+                'item/commonInteractions/extendedTextInteraction/hasMath' => [
+                    'taoQtiItem/qtiCreator/widgets/interactions/extendedTextInteraction/states/Question' => ['hasMath'],
+                ],
+                'item/commonInteractions/hottextInteraction/disallowHTMLInHottext' => [
+                    'taoQtiItem/qtiCreator/widgets/interactions/hottextInteraction/states/Question'=> ['disallowHTMLInHottext'],
+                ]
+            ]
+        ],
+        'taoQtiItem' => [
+            'qtiCreator' => [
+                'item/multiColumn' => [
+                    'multi-column' => []
+                ],
+                'item/scrollableMultiColumn' => [
+                    'scrollable-multi-column' => []
+                ],
+                'item/perInteractionRp' => [
+                    'perInteractionRp' => []
+                ],
+            ]
+        ]
+    ];
+
+    public function up(Schema $schema): void
+    {
+        $resultConfig = self::DEFAULT_CONFIG;
+        $backOfficeExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById('taoBackOffice');
+        if($backOfficeExtension === null) {
+            throw new \Exception(sprintf('Cannot find %s extension', 'taoBackOffice'));
+        }
+        $existingConfig = $backOfficeExtension->getConfig(self::GLOBAL_UI_CONFIG_NAME);
+        if($existingConfig !== false) {
+            $resultConfig = $existingConfig;
+        }
+
+        foreach(self::EXISTING_CONFIG_MAP as $extensionName => $extensionConfig) {
+            $currentExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById($extensionName);
+            if($currentExtension === null) {
+                throw new \Exception(sprintf('Cannot find %s extension', $extensionName));
+            }
+            foreach($extensionConfig as $configFileName => $existingConfig) {
+                $currentExtensionConfig = $currentExtension->getConfig($configFileName);
+                if($currentExtensionConfig === false) {
+                    throw new \Exception(sprintf('Cannot find %s config', $configFileName));
+                }
+
+                $this->updateGlobalUIConfig($existingConfig, $currentExtensionConfig,$resultConfig);
+            }
+        }
+
+        $backOfficeExtension->setConfig(self::GLOBAL_UI_CONFIG_NAME, $resultConfig);
+    }
+
+    private function updateGlobalUIConfig($existingConfig, $currentExtConfig, array &$resultConfig): void
+    {
+        foreach($existingConfig as $globalConfigKey => $configData) {
+            foreach($configData as $configKey => $singleConfig) {
+                if(!array_key_exists($configKey, $currentExtConfig)) {
+                    continue;
+                }
+                if (!empty($singleConfig)) {
+                    foreach($singleConfig as $value) {
```

Comment:
I'm a bit concerned about the number of loops and conditions. Perhaps just grabbing them manually is easier. Wdyt?

```
$configs = array_merge($configs, $this->getFromTaoCore());
$configs = array_merge($configs, $this->getFromItemQti());

...

private function getFromTaoCore(): array
{
    $currentExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById('tao');
    
    if ($currentExtension === null) {
        return [];
    }

    $configs = $currentExtension->getConfig('client_lib_config_registry');

    return [
        'item/image/mediaAlignment' => $this->visibilityAsString($configs['ui/image/ImgStateActive']['mediaAlignment'] ?? false),
        'test/item/BRS' => $this->visibilityAsString($configs['taoQtiTest/controller/creator/views/item']['BRS'] ?? false),
         // ...
    ];
}

private function visibilityAsString(bool $isVisible): string
{
    return $isVisible ? self::SHOW_OPTION : self::HIDE_OPTION;
}
```

Reply:
Great idea! Refactored according to your the proposal